v10.9 Changelog (#7954)
mattermost/docs: source/about/mattermost-v10-changelog.md

```{Important}
```{include} common-esr-support-upgrade.md
```
(release-v10.9-feature-release)=
## Release v10.9 - [Feature Release](https://docs.mattermost.com/about/release-policy.html#release-types)

**Release day: 2025-06-16**

### Compatibility
 - Updated the minimum supported versions of Edge and Chrome to 134+, and Firefox to 128+.

### Important Upgrade Notes
 - A new index to the ``CategoryId`` column in ``SidebarChannels`` table was added to improve query performance. No database downtime is expected for this upgrade. See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) for more details.

```{Important}
If you upgrade from a release earlier than v10.8, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
```

### Highlights

#### New Enterprise Advanced License
 - Added support for a new Enterprise Advanced license. This new license is supported starting v10.9 and is supported on PostgreSQL only. Enterprise plugins need to be updated to support the new license (most of which are pre-packaged in v10.9).

### Improvements

#### User Interface (UI)

>  - Consolidated all channel editing functionalities into a single, accessible modal located in the channel header menu. Users can now update channel names, URL slugs, convert between public and private, modify/add a purpose and header (with a live markdown preview), manage channel banners, and archive the channel—all in one place. Updates include safeguards for unsaved edits, improved URL-slug editing, and enhanced keyboard and navigation accessibility.

Reviewer: ```suggestion
 - Consolidated all channel editing functionality into a single, accessible modal located in the channel header menu. Users can now update channel names, URL slugs, convert between public and private, modify/add a purpose and header (with a live markdown preview), manage channel banners, and archive the channel—all in one place. Updates include safeguards for unsaved edits, improved URL-slug editing, and enhanced keyboard and navigation accessibility.
```

 - Improved the **Drafts** list by implementing virtualization.
 - Enhanced the behavior for reporting issues in the platform.
 - Introduced minor layout tweaks in theme settings for improved usability.

#### Administration
 - Added support for attribute based channel access (disabled and behind a feature flag).
 - Added support for AES-256-GCM encryption in SAML.
 - Updated the email validation logic to ensure Mattermost no longer accepts email addresses enclosed in angle brackets (e.g., ``<billy@example.com>``). Although these comply with RFC 5322 and RFC 6532 standards, they introduce unnecessary complexity. If a user already has such an email in your installation, they may face issues like being unable to update their profile. To resolve this, the email must be modified manually using the command: ``mmctl user email "<affecteduser@domain.com>" affecteduser@domain.com``.
 - Added a license load metric to the **About** screen to display current license usage.
 - Upgraded the logr dependency to version 2.0.22.
 - Removed telemetry tracking from Redux selectors.
 - Made it possible to view JSON logs in plain text within the **System console**.

>  - Enhanced the **System console** search functionality to include all log fields.

Reviewer 2: ```suggestion
 - Enhanced the **System Console** search functionality to include all log fields.
```

 - Enhanced error reporting related to cluster management.
 - Added a LDAP setting to re-add removed members.
 - Added support for SSO while Mattermost is embedded in an iframe.

#### Performance
 - Optimized the team-switching operation by eliminating unnecessary calls to retrieve channels and channel members.
 - Improved websocket re-opening speed when network conditions change.

### Bug Fixes

>  - Fixed various issues on the "Create Team" screen.

Reviewer 2: ```suggestion
 - Fixed various issues on the **Create Team** screen.
```

 - Fixed several accessibility issues across the login process, account creation, and MFA setup.
 - Fixed an issue where horizontal rule (HR) elements were not visible in preview mode in the right-hand sidebar (RHS).
 - Fixed an issue with inconsistent sizing of markdown images in preview mode.
 - Fixed a keyboard navigation issue within thread items.
 - Fixed layout issues with the emoji picker on mobile browsers.
 - Fixed an issue with the positioning of “edited” text and tooltips in certain scenarios.
 - Fixed the accessibility of the search box.

>  - Fixed an issue where post list scrolling didn’t work when pressing the "Page Up" or "Page Down" keys.

Reviewer 2: ```suggestion
 - Fixed an issue where post list scrolling didn’t work when pressing the **Page Up** or **Page Down** keys.
```

### Important Upgrade Notes
 - A new index to the ``CategoryId`` column in ``SidebarChannels`` table was added to improve query performance. No database downtime is expected for this upgrade. See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) for more details.
 - Schema changes in the form of a new materialized view (``AttributeView``) was added that aggregates user attributes into a separate table. No database downtime is expected for this upgrade. See the [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html) for more details.

```{Important}
If you upgrade from a release earlier than v10.8, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
```

### Highlights

#### New Enterprise Advanced License
 - Added support for a new Enterprise Advanced license. This new license is supported starting v10.9 and is supported on PostgreSQL only. Enterprise plugins need to be updated to support the new license (most of which are pre-packaged in v10.9).

### Improvements

#### User Interface (UI)

>  - Consolidated all channel editing functionality into a single, accessible modal located in the channel header menu. Users can now update channel names, URL slugs, convert between public and private, modify/add a purpose and header (with a live markdown preview), manage channel banners, and archive the channel—all in one place. Updates include safeguards for unsaved edits, improved URL-slug editing, and enhanced keyboard and navigation accessibility.

Reviewer 3: We actually can't convert to public from private so we should update to: 

```suggestion
 - Consolidated all channel editing functionality into a single, accessible modal located in the channel header menu. Users can now update channel names, URL slugs, convert to private, modify/add a purpose and header (with a live markdown preview), manage channel banners, and archive the channel—all in one place. Updates include safeguards for unsaved edits, improved URL-slug editing, and enhanced keyboard and navigation accessibility.
```

 - Pre-packaged MS Teams plugin [v2.2.1](https://github.com/mattermost/mattermost-plugin-msteams/releases/tag/v2.2.1).
 - Pre-packaged Playbooks plugin [v2.2.0](https://github.com/mattermost/mattermost-plugin-playbooks/releases/tag/v2.2.0).
 - Pre-packaged Calls plugin [v1.8.0](https://github.com/mattermost/mattermost-plugin-calls/releases/tag/v1.8.0).
 - Pre-packaged Jira plugin version [v4.3.0](https://github.com/mattermost/mattermost-plugin-jira/releases/tag/v4.3.0).
 - Introduced a configurable channel banner feature for channel admins, visible across desktop, web, and mobile platforms. This feature requires an Enterprise Advanced license.
 - Added more descriptive page titles to the login, account creation, and password reset pages.
 - Improved the **Drafts** list by implementing virtualization.
 - Enhanced the behavior for reporting issues in the platform.
 - Introduced minor layout tweaks in theme settings for improved usability.

#### Administration

>  - Added support for attribute based channel access (disabled and behind a feature flag).

Reviewer 3: ```suggestion
 - Added support for attribute based channel access for Enterprise Advanced licensed servers (currently behind a feature flag ).
```

```{Important}
If you upgrade from a release earlier than v10.8, please read the other [Important Upgrade Notes](https://docs.mattermost.com/upgrade/important-upgrade-notes.html).
```

### Improvements

#### User Interface (UI)
 - Consolidated all channel editing functionalities into a single, accessible modal located in the channel header menu. Users can now update channel names, URL slugs, convert between public and private, modify/add a purpose and header (with a live markdown preview), manage channel banners, and archive the channel—all in one place. Updates include safeguards for unsaved edits, improved URL-slug editing, and enhanced keyboard and navigation accessibility.
 - Pre-packaged MS Teams plugin [v2.2.1](https://github.com/mattermost/mattermost-plugin-msteams/releases/tag/v2.2.1).
 - Pre-packaged Playbooks plugin [v2.2.0](https://github.com/mattermost/mattermost-plugin-playbooks/releases/tag/v2.2.0).
 - Pre-packaged Calls plugin [v1.8.0](https://github.com/mattermost/mattermost-plugin-calls/releases/tag/v1.8.0).
 - Pre-packaged Jira plugin version [v4.3.0](https://github.com/mattermost/mattermost-plugin-jira/releases/tag/v4.3.0).

>  - Introduced a configurable channel banner feature for channel admins, visible across desktop, web, and mobile platforms. This feature requires an Enterprise Advanced license.

Reviewer 4: Should we separately announce the Enterprise Advanced License? When we do, can we mention that it is Postgres-only?

Reviewer 3: @cwarnermm Do we have anywhere in the documentation that we can outline some requirements for the new sku? We need to highlight: 

- From which MM Version is supported (v10.9)
- Highlight that enterprise plugins would need to be updated to support the. new license as well (most prepackaged in that version) 
- Database requirement.

Author: Added a line in the "Administration" section for the new SKU.

 - Under ``ExperimentalAuditSettings`` in ``config.json``:
    - Added ``Certificate`` configuration setting to accept a certificate to be used for audit logging egress.
 - Under ``LdapSettings`` in ``config.json``:
    - Added ``ReAddRemovedMembers`` configuration setting to add a LDAP setting to re-add removed members.

### Open Source Components
 - Added ``monaco-editor`` and ``monaco-editor-webpack-plugin``, and removed ``dynamic-virtualized-list``, ``popper.js``, ``react-hot-loader``, ``react-popper`` from https://github.com/mattermost/mattermost.

### Go Version
 - v10.9 is built with Go ``v1.23.7``.

### Known Issues
 - Setting the license file location through an environment variable still gives the option to upload a new license through the System Console, resulting in the license being overwritten by the one set through the environment variable. See this [knowledge base article](https://support.mattermost.com/hc/en-us/articles/33911983851284-System-console-still-displays-old-license-after-uploading-a-new-one) on how to resolve this issue.
 - Searching stop words in quotation marks with Elasticsearch enabled returns more than just the searched terms.

>  - Slack import through the CLI fails if email notifications are enabled.

Reviewer 4: Can you remind me of the criteria for including known issues in this list? I feel like our entire open backlog could qualify here, and the ones listed here don't feel particularly high-impact.

Should we just remove this section?

Author: It's normally for new regressions in a release that were not fixed in time. I've removed everything else expect the first known issue as it was previously requested to be added by the CS team.
